src/gsi/setupoz.f90

```diff
@@ -410,21 +417,21 @@ subroutine setupozlay(obsLL,odiagLL,lunin,mype,stats_oz,nlevs,nreal,nobs,&
            call grdcrd1(ozp_omi(nloz_omi),prsitmp,nsig+1,-1)
         end if
         
-        if (obstype /= 'omieff' .and. obstype /= 'tomseff') then
+        call tintrp2a1(ges_oz,ozgestmp,dlat,dlon,dtime,hrdifsig,&
+          nsig,mype,nfldsig)
+
+        ! need call to get pressures for pressure level output in ncdiags
+        call tintrp2a1(ges_prsi,prsitmp,dlat,dlon,dtime,hrdifsig,&
+          nsig+1,mype,nfldsig)
+
+
+        if (obstype /= 'omieff' .and. obstype /= 'tomseff' .and. &
+            obstype /= 'ompsnmeff' ) then
            call intrp3oz1(ges_oz,ozges,dlat,dlon,ozp,dtime,&
                 nlevs,mype,doz_dz)
         endif
 
-!       Check scan position errors in ompstc8
```

Comment:
I believe this section here is to remove the bad quality omps total column obs. This section needs to be kept